### ISSUE TITLE
Open platform proposal

### ISSUE THREAD

**Issue by brunagravalos** [brunagravalos/EXPECT_personal#3]
## Components of the platform

The platform is composed of:

1. A GitHub repository containing datasets, models, benchmarks, tutorials, and guideline files.
2. An MkDocs Material website that renders all content into a structured, searchable, and professional documentation site.
3. Integrations:
   - HuggingFace Hub for models and datasets

## MkDocs Material
The website organizes content into major sections:
```
Home
Datasets
Models
Benchmarks
Tutorials
Guidelines
```
MkDocs generates the site  from the contents of the docs/ folder.

## Repository Structure

```
ml-platform/
 ├── datasets/
 │     ├── dataset1/
 │     │      ├── README.md
 │     │      ├── download.py
 │     │      └── metadata.yaml
 │     └── dataset2/
 ├── models/
 │     ├── model1/
 │     │      ├── README.md
 │     │      ├── config.json
 │     │      └── hub_link.txt
 │     └── model2/
 ├── benchmarks/
 │     ├── task1/
 │     │      ├── README.md
 │     │      ├── metric.py
 │     │      ├── evaluate.py
 │     │      └── results.md
 │     └── task2/
 ├── tutorials/    
 │     └── notebooks??
 ├── guidelines/
 │     └── benchmarking.md
 └──docs/
```
### Datasets
Datasets are not stored directly in the repository. Instead, each dataset folder contains:
- A Markdown description (README.md)
- A Python download script (download.py)
- Structured metadata (metadata.yaml)

They could be stored in Hugging Face (need to investigate better but seems like there is simple code to download).

```
from datasets import load_dataset
ds = load_dataset("username/dataset-name")
```
#### Benchmarks
Benchmark = task + dataset + metric + evaluation code.
Inside benchmarks/task1/ should go:

**README.md**

A description:
- task definition
- dataset used
- metric used
- baseline model
- reproducibility instructions

**scrpt.py**

A script that:
- loads dataset
- loads model
- computes metric
- prints score

#### Models

```
models/
   example_model/
      README.md            ← Model documentation similar to the one in MkDocs
      config.json          ← Architecture/hyperparameters
      hub_link.txt         ← Link to the Hugging Face repo
      example.py           ← Example inference script
```

For each model, I create a repository on Hugging Face:
Inside the Hugging Face model repo:

- Model
- A model card README.md

To load models (or similar, to do actually try it): 

```
from huggingface_hub import hf_hub_download
import torch

model_path = hf_hub_download(
    repo_id="consortium-org/resnet50",
    filename="pytorch_model.bin"
)

state_dict = torch.load(model_path)
model.load_state_dict(state_dict)

```
**Versioning:**
You can create Git-style tags like:
```
v1
v1.1
v2
final-2025
baseline
```
Where tags point to a specific commit.
To load a specific version:

`model = AutoModel.from_pretrained("org/model-name", revision="v1")`
